src/Datadog.Trace/Propagation/PropagationHelpers.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.Conventions;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Propagation
+{
+    internal static class PropagationHelpers
+    {
+        public static TraceId ParseTraceId<T>(T carrier, Func<T, string, IEnumerable<string>> getter, string headerName, ITraceIdConvention traceIdConvention, IDatadogLogger logger)
+        {
+            var headerValues = getter(carrier, headerName).ToList();
```

Comment:
Do we need the call to `ToList()` here?

Reply:
should be good with deferred execution as well - might gain actually some performance than just an allocation win.